support/ruby/consolr/spec/consolr_spec.rb

```diff
@@ -0,0 +1,133 @@
+require 'spec_helper'
+
+describe Consolr::Console do
+
+  before (:each) do
+    @console = Consolr::Console.new
+  end
+
+  it 'fails if no hostname is provided' do
+    expect { @console.start( {:hostname => nil}) }.to raise_error(SystemExit, /asset tag or hostname/)
+  end
+
+  it 'fails if both tag and hostname are passed' do
+    expect { @console.start({:hostname => 'host.dc.net', :tag => '001234' }) }.to raise_error(SystemExit, /not both/)
+  end
+
+  it 'fails if tag cannot be found' do
+    expect { @console.start({:tag => 'bogus_tag'}) }.to raise_error(SystemExit, /Found 0 assets/)
+  end
+
+  it 'fails if hostname cannot be found' do
+    expect { @console.start({:hostname => 'bogus_hostname'}) }.to raise_error(SystemExit, /Found 0 assets/)
+  end
+
+  it 'fails if multiple hosts are found' do
+    expect { @console.start({:hostname => 'hostname-with-multiple-assets.dc.net'}) }.to raise_error(SystemExit, /Found \d+ assets/)
+  end
+
+  safe_boolean_actions = {:console => "--> Opening SOL session (type ~~. to quit)\nsol activate", :kick => 'sol deactivate', :identify => 'chassis identify', :sdr => 'sdr elist all', :on => 'power on'}
+  dangerous_boolean_actions = {:off => 'power off', :reboot => 'power cycle'}
+
+  describe 'safe allocated asset' do
+    safe_allocated_tag = 'safe-allocated-tag'
+    safe_boolean_actions.each do |action, response|
+      it "does #{action}" do
```

Comment:
perhaps adding "safe" in the it description would be clearer and symmetric with how danger is done.